lib/OnyxConnectionManager.ts

```diff
@@ -0,0 +1,266 @@
+import bindAll from 'lodash/bindAll';
+import * as Logger from './Logger';
+import type {ConnectOptions} from './Onyx';
+import OnyxUtils from './OnyxUtils';
+import * as Str from './Str';
+import type {DefaultConnectCallback, DefaultConnectOptions, OnyxKey, OnyxValue} from './types';
+
+type ConnectCallback = DefaultConnectCallback<OnyxKey>;
+
+/**
+ * Represents the connection's metadata that contains the necessary properties
+ * to handle that connection.
+ */
+type ConnectionMetadata = {
+    /**
+     * The subscription ID returned by `OnyxUtils.subscribeToKey()` that is associated to this connection.
+     */
+    subscriptionID: number;
+
+    /**
+     * The Onyx key associated to this connection.
+     */
+    onyxKey: OnyxKey;
+
+    /**
+     * Whether the first connection's callback was fired or not.
+     */
+    isConnectionMade: boolean;
+
+    /**
+     * A map of the subcriber's callbacks associated to this connection.
+     */
+    callbacks: Map<string, ConnectCallback>;
+
+    /**
+     * The last callback value returned by `OnyxUtils.subscribeToKey()`'s callback.
+     */
+    cachedCallbackValue?: OnyxValue<OnyxKey>;
+
+    /**
+     * The last callback key returned by `OnyxUtils.subscribeToKey()`'s callback.
+     */
+    cachedCallbackKey?: OnyxKey;
+};
+
+/**
+ * Represents the connection object returned by `Onyx.connect()`.
+ */
```

Comment:
Same as above

Reply:
I think this comment got shifted to the wrong line... same as this: https://github.com/Expensify/react-native-onyx/pull/567#discussion_r1723128171

my bad sorry